store/sqlstore/thread_store.go

```diff
@@ -1010,12 +1061,12 @@ func (s *SqlThreadStore) DeleteOrphanedRows(limit int) (deleted int64, err error
 // return number of unread replies for a single thread
 func (s *SqlThreadStore) GetThreadUnreadReplyCount(threadMembership *model.ThreadMembership) (unreadReplies int64, err error) {
 	query, args, err := s.getQueryBuilder().
-		Select("COUNT(Posts.Id)").
-		From("Posts").
+		Select("(Threads.ReplyCount - ThreadMemberships.SeenReplyCount)").
+		From("ThreadMemberships").
+		InnerJoin("Threads ON ThreadMemberships.PostId = Threads.PostId").
```

Comment:
Will an `INNER JOIN` result in a `ErrNotFound` error if for some reason the corresponding `Threads` row doesn't exist? (vs. `0` before)